### PR TITLE
Express notes api

### PR DESCRIPTION
**Pull request by rskidmore1**
image

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+const notes = require('./data')
+
+const app = express();
+
+function write(input){
+  fs.writeFile('data.json', input, (err) => {
+    if (err) throw err;
+    console.log('File write successful');
+  });
+}
```
Comment (thebearingedge) [inline express-notes-api/index.js:12]: I don't think you need this function anymore.

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  res.status(200).json(notesSend);
+})
+
+
+app.get('/api/notes/:id', (req, res, next) => {
+
+  if (Number.isInteger(Number(req.params.id))  && Number(req.params.id) > 0){
+
+
+    if (String(req.params.id) in notes.notes){
```
Comment (thebearingedge) [inline express-notes-api/index.js:30]: all `req.params` properties are strings, so you can leave out this `String()` coercion. Check the rest of the code for it too.

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+
+})
+
+
+app.put('/api/notes/:id', (req,res) =>{
+
+  if ((Number.isInteger(Number(req.params.id)) === false && Number(req.params.id) < 0) || req.body.content === undefined){
+    res.status(404).json({'error': 'Invalid input'});
+  } else if ((Number.isInteger(Number(req.params.id))  && Number(req.params.id) > 0) && req.body.content !== undefined){
```
Comment (thebearingedge) [inline express-notes-api/index.js:96]: If you store the coerced id in a variable, you don't have to repeat `Number(req.params.id)` all over the place.

```js
const id = Number(req.params.id);
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Good, just a couple minor notes.

**Review by thebearingedge — APPROVED** [review]

**Comment by rskidmore1** [comment]
muchthanks

**Pull request merged**